src/middleware/basicAuthentication.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response, NextFunction } from 'express';
+import bcrypt from 'bcrypt';
+import UserService from '../user/UserService';
+import ForbiddenException from '../errors/ForbiddenException';
+
+// middleware to check if user is authorized
+const basicAuthentication = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Middleware function for basic authentication and if user is authenticated, it will add the user to req.user variable